### ISSUE TITLE
Validator upgrade vote did not work

### ISSUE THREAD

**Issue by agouin** [OLSF/libra#865]
# 🐛 Bug

https://0l.interblockcha.in/tx/8145525

This vote appeared successful from the command line and from the blockchain explorer, but it did not show up as a vote in the upgrades section of the web-monitor. After trying again two hours later, it successfully appeared in the web-monitor.

https://0l.interblockcha.in/tx/8166126

## To reproduce
```bash
txs oracle-upgrade -v -h 4fc1070c0a7442cb14949b5bc6e534e7c0c2daeee57da03c479b21a49975f418
```

**Stack trace/error message**

No error

## Expected Behavior

Transaction shows error, or vote appears in list of voters for upgrade (included in count in web explorer).

## System information

**Please complete the following information:**
- ol 0.1.0 (6d7d7d12bcc8984a350b765ac24ab643e6351655)
- rustc 1.56.1 (59eed8a2a 2021-11-01)
- Ubuntu 20.04 with kernel 5.11.0

**Comment by dboreham** [comment]
Clarification: this is a suspected bug in the web monitor, yes?

**Comment by agouin** [comment]
I don't think so since the transaction did not have an error on the chain. It appears to be deeper than the web monitor.

**Comment by dboreham** [comment]
Hmm. So the bug is the transaction succeeded when it should have failed? (sorry if I'm being slow -- been eating turkey for 24h).

**Comment by 0o-de-lally** [comment]
@agouin were you a current validator in the validator set?

**Comment by agouin** [comment]
@dboreham I think it should have either failed with a Move Abort and an error message for why it failed, or it should have shown up in the list since it has a success message [8145525](https://0l.interblockcha.in/block/8145525)

@0o-de-lally yes, I was onboarded in epoch 29, joined the validator set in the first hour of epoch 30, and the blocks mentioned here were during epoch 31
